build/commands/lib/config.js

```diff
@@ -303,6 +304,19 @@ Config.prototype.buildArgs = function () {
 
   if (!this.isBraveReleaseBuild()) {
     args.chrome_pgo_phase = 0
+
+    if (process.platform === 'darwin' && this.targetOS != 'ios' && !this.shouldSign()) {
+      // Currently we're using is_official_build mode in PR builds on CI. This enables dSYMs
+      // by default, which slows down link phase, but also disables relocatable compilation 
+      // on MacOS (aka 'zero goma cachehits' style).
+      //
+      // Don't create dSYMs in non-public Release builds and don't strip debug info.
+      // See //build/config/apple/symbols.gni for additional details.
+      if (!this.isAsan()) {
+        args.enable_dsyms = false
+      }
+      args.enable_stripping = false
```

Comment:
This basically reverts what `is_official_build` is setting inside `gn` scripts. This is currently required to have relocatable compilation on Mac, i.e. goma doesn't work in `Release` build configuration without disabling dSYMs.

Reply:
@goodov this is not the fix we want for this, @atuchin-m is working on fixing this inside gn based on the new release flag

Reply:
actually I'm not following why we're disabling stripping here and basing it on `!shouldSign` doesn't make sense to me because we sometimes run signed PR builder builds

Reply:
the point is that `shouldSign` is not a good way to differentiate PR builder from release builds because we sometimes do run signed PR builds. `is_brave_release_build` is how we differentiate, but see slack thread because I think we're making this more complicated than it needs to be by running PR builder with `is_official_build` in the first place

Reply:
I’ve removed `enable_stripping`, because it was not actually necessary, but kept `enable_dsyms=false` under `isOfficialBuild` so goma can work now in PRs and local builds.